src/test/java/io/nats/client/OptionsTests.java

```diff
@@ -688,6 +710,24 @@ public void testInboxPrefixCoverage() {
         assertEquals("foo.", o.getInboxPrefix());
     }
 
+    @Test
+    public void testSslContextIsProvided() {
+        Options o = new Options.Builder().server("nats://localhost").build();
+        assertNull(o.getSslContext());
+        o = new Options.Builder().server("ws://localhost").build();

```

Comment:
Open question here: Do we want to support non-SSL websockets? In other words, if the url begins with "ws://" do we want to disable TLS? As it is now "ws://" is synonymous with "wss://"... and do we want to support non-secure "ws://" which is "upgrade"d to TLS using the NATS upgrade to secure mechanism?

Reply:
@ColinSullivan1 @RichardHightower Any opinions?